src/gleam_stdlib.mjs

```diff
@@ -344,18 +344,7 @@ export function power(base, exponent) {
 }
 
 export function random_uniform() {
-  const random_uniform_result = Math.random();
-  // With round-to-nearest-even behavior, the ranges claimed for the functions below
-  // (excluding the one for Math.random() itself) aren't exact.
-  // If extremely large bounds are chosen (2^53 or higher),
-  // it's possible in extremely rare cases to calculate the usually-excluded upper bound.
-  // Note that as numbers in JavaScript are IEEE 754 floating point numbers
-  // See: <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random>
-  // Because of this, we just loop 'until' we get a valid result where 0.0 <= x < 1.0:
-  if (random_uniform_result === 1.0) {
-    return random_uniform();
-  }
-  return random_uniform_result;
```

Comment:
The note itself says that this error may occur "excluding the one for Math.random() itself". So we don't actually need the check here. The other `random` functions we have might encounter this issue.

I'm not sure if we should warn this in the documentation, since it is a really really obscure issue only if you have a ridiculously large range (but it would be frustrating to encounter it).

Reply:
I am strongly against this change. The stdlib should be about soundness and correctness. It can simply happen that it is 1.0. Anyone can build a faster ffi themselves that behaves incorrect in very unlikely siutations.

Reply:
`Math.random()` will never return 1,0

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random#examples
"the ranges claimed for the functions below (excluding the one for Math.random() itself) aren't exact"
Thus `Math.random()` will have an exact range.

Although that brings up the point that we could do the check in float.random and int.check, which could encounter this issue.

Reply:
I've made a bunch of changes, with tests to check for the rounding error. 

The checks for `float.random` are in both the erlang and JS targets. `int.random` has this check for the JS target. (The erlang target doesn't need it because it uses `rand:uniform` for ints).